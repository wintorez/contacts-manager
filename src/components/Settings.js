import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Header, Form, Checkbox, Button } from 'semantic-ui-react';
import Page from './Page';

const Settings = ({ location }) => (
  <Page title="Settings" location={location}>
    <Header as="h3">Settings</Header>
    <Form>
      {['1', '2', '3', '4'].map(number => (
        <Form.Field key={number}>
          <Checkbox label={`Value ${number}`} />
        </Form.Field>
      ))}
      <Button type="submit">Save</Button> or <Link to="/employees">Cancel</Link>
    </Form>
  </Page>
);

Settings.propTypes = {
  location: PropTypes.object.isRequired
};

export default Settings;
