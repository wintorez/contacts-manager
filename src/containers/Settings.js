import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Settings from '../components/Settings';

const mapStateToProps = (state, ownProps) => ({});

const mapDispatchToProps = (dispatch, ownProps) => ({});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Settings)
);
