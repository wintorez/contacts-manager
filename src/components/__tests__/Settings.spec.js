import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Settings from '../Settings';

const defaults = {
  history: {},
  location: { pathname: '/settings' }
};

describe('<Settings />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Settings {...defaults} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
