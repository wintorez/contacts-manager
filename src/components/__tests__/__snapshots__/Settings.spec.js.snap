// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> renders correctly 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui tabular menu"
  >
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/employees"
        onClick={[Function]}
      >
        Employees
      </a>
    </div>
    <div
      className="active item"
      onClick={[Function]}
    >
      <a
        href="/settings"
        onClick={[Function]}
      >
        Settings
      </a>
    </div>
  </div>
  <div
    className="ui container"
  >
    <h3
      className="ui header"
    >
      Settings
    </h3>
    <form
      action={undefined}
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <div
          className="ui checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            disabled={undefined}
            id={undefined}
            name={undefined}
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
            value={undefined}
          />
          <label
            htmlFor={undefined}
          >
            Value 1
          </label>
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="ui checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            disabled={undefined}
            id={undefined}
            name={undefined}
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
            value={undefined}
          />
          <label
            htmlFor={undefined}
          >
            Value 2
          </label>
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="ui checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            disabled={undefined}
            id={undefined}
            name={undefined}
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
            value={undefined}
          />
          <label
            htmlFor={undefined}
          >
            Value 3
          </label>
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="ui checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            disabled={undefined}
            id={undefined}
            name={undefined}
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
            value={undefined}
          />
          <label
            htmlFor={undefined}
          >
            Value 4
          </label>
        </div>
      </div>
      <button
        className="ui button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
        type="submit"
      >
        Save
      </button>
       or 
      <a
        href="/employees"
        onClick={[Function]}
      >
        Cancel
      </a>
    </form>
  </div>
</div>
`;
